### PR TITLE
Improve mempool_package_limits.py

### PR DESCRIPTION
**Pull request by naiza2000**
Follow-up to #21800 adding changes suggested in https://github.com/bitcoin/bitcoin/pull/21800#pullrequestreview-725180293

### CODE REVIEW

File: test/functional/test_framework/wallet.py [hunk 1]
```diff
@@ -182,28 +182,30 @@ def sendrawtransaction(self, *, from_node, tx_hex):
...
-        "vout": n,
+        "vout": 0,
         "scriptPubKey": parent_locking_script,
         "amount": parent_value,
     }] if parent_locking_script else None
     signedtx = node.signrawtransactionwithkey(hexstring=rawtx, privkeys=privkeys, prevtxs=prevtxs)
     assert signedtx["complete"]
     tx = tx_from_hex(signedtx["hex"])
-    return (tx, signedtx["hex"], my_value, tx.vout[0].scriptPubKey.hex())
+    return tx, signedtx["hex"], my_value, tx.vout[0].scriptPubKey.hex()
```
Comment (glozow) [inline test/functional/test_framework/wallet.py:203]: I think https://github.com/bitcoin/bitcoin/pull/21800#discussion_r685061119 suggested changing this return value to a named tuple instead?

File: test/functional/test_framework/wallet.py [hunk 2]
```diff
@@ -238,18 +240,18 @@ def bulk_transaction(tx, node, target_weight, privkeys, prevtxs=None):
     """Pad a transaction with extra outputs until it reaches a target weight (or higher).
     returns CTransaction object
     """
-    tx_heavy = deepcopy(tx)
```
Comment (glozow) [inline test/functional/test_framework/wallet.py:241]: Can also remove the import as well, iirc

File: test/functional/test_framework/wallet.py [hunk 3]
```diff
@@ -182,28 +182,30 @@ def sendrawtransaction(self, *, from_node, tx_hex):
         from_node.sendrawtransaction(tx_hex)
         self.scan_tx(from_node.decoderawtransaction(tx_hex))
 
-def make_chain(node, address, privkeys, parent_txid, parent_value, n=0, parent_locking_script=None, fee=DEFAULT_FEE):
-    """Build a transaction that spends parent_txid.vout[n] and produces one output with
+def make_chain(node, address, privkeys, parent_txid, parent_value, parent_locking_script=None, fee=DEFAULT_FEE):
```
Comment (glozow) [inline test/functional/test_framework/wallet.py:185]: https://github.com/bitcoin/bitcoin/pull/21800/files#r685091478 also suggested renaming this function

I'd suggest something like `create_child` ?

**Review by glozow — COMMENTED** [review]
Concept ACK, thanks for working on this

Linking the suggestions in the PR description and/or describing the changes within the commit messages would help reviewers understand what the improvements are. Also, you seem to have a merge commit that should be squashed?

File: test/functional/rpc_packages.py [hunk 4]
```diff
@@ -133,7 +133,7 @@ def test_independent(self):
     def test_chain(self):
         node = self.nodes[0]
         first_coin = self.coins.pop()
-        (chain_hex, chain_txns) = create_raw_chain(node, first_coin, self.address, self.privkeys)
+        chain_hex, chain_txns = create_raw_chain(node, first_coin, self.address, self.privkeys)
```
Comment (glozow) [inline test/functional/rpc_packages.py:136]: You need to add the chain length here now that the default has been removed

**Comment by jnewbery** [comment]
Concept ACK. Will review once @glozow's comments have been addressed and the CI failures are fixed.

**Comment by glozow** [comment]
@naiza2000 still working on this?

**Comment by naiza2000** [comment]
> @naiza2000 still working on this?

Yes, I was out of town for the last few days, I have started working on it again.

**Comment by glozow** [comment]
Needs rebase and squash?

**Comment by MarcoFalke** [comment]
Can be closed?

**Comment by glozow** [comment]
> Can be closed?

Seems so. I can open a PR to clean up all the followups from #21800

**Comment by fanquake** [comment]
> Seems so. I can open a PR to clean up all the followups from https://github.com/bitcoin/bitcoin/pull/21800

@glozow did this happen? Can we remove up for grabs?

**Pull request closed without merging**